CMakeLists.txt

```diff
@@ -85,6 +85,22 @@ add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/Source/Tools)
 add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/Source/Tools/alive_api)
 add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/assets)
 
+# Setup cross-platform PCH
+if(NOT MSVC)
+    target_precompile_headers(
+        AliveLibCommon PRIVATE
+        "${CMAKE_CURRENT_SOURCE_DIR}/Source/AliveLibCommon/pch_shared.h"
+    )
+
+    target_precompile_headers(AliveLibAO REUSE_FROM AliveLibCommon)
+    target_precompile_headers(AliveLibAE REUSE_FROM AliveLibCommon)
+    target_precompile_headers(AliveExe REUSE_FROM AliveLibCommon)
+    target_precompile_headers(AliveExeAO REUSE_FROM AliveLibCommon)
+    target_precompile_headers(AliveExeAE REUSE_FROM AliveLibCommon)
+    # `vab_tool`, `alive_api`, and `alive_api_test` targets cannot make use of PCH
+    # due to `CMAKE_CXX_EXTENSIONS` being set to `OFF`

```

Comment:
is it worth turning it on there? also could this replace the msvc specific PCH handling?

Reply:
> is it worth turning it on there?

Possibly, yes. I can run some tests.

---

> could this replace the msvc specific PCH handling?

I think so, but I do not use MSVC so I can't really test it.